pkg/cephfs/nodeserver.go

```diff
@@ -167,11 +169,15 @@ func (ns *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 		return nil, err
 	}
 
-	// Configuration
-
 	targetPath := req.GetTargetPath()
 	volID := req.GetVolumeId()
 
+	if acquired := ns.VolumeLocks.TryAcquire(targetPath); !acquired {
```

Comment:
Should this conflict with Staging operation? I would say it is safer it it does, as the assumed need for this is due to the CO sending multiple calls due to state loss, hence lock on the `VolID` instead.
It would serialize parallel publish requests of the same volume to different targets, but metrics show this is a fast operation on the local node, and hence should not suffer much due to the serialization slowness. (Unpublish would also need to change to the `VolID`)